sockets/twitch.js

```diff
@@ -0,0 +1,160 @@
+        twitchPollingInterval = setInterval(
+          function() {
+            getTwitchPollableData(twitchPollCache)
+          }, twitchPollFrequency);
+      }
+      else
+      {
+        console.log('Stopping Twitch polling...');
+        clearInterval(twitchPollingInterval);
+      }
+
+      twitchIO.emit('update twitch', twitchData);
+      console.log('update twitch: ' + JSON.stringify(twitchData));
+    });
+
+    socket.on('request twitch', function() {
+      console.log('fetching old Twitch data: ' + JSON.stringify(data));
+      socket.emit('update twitch', twitchData);
+    });
+
+    function getTwitchPollableData(twitchData) {
+      console.log('Polling Twitch for updates...');
+    	twitchData = getTwitchFollowerData(twitchData);
+      twitchData = getTwitchViewerData(twitchData);
+      console.log('New Twitch Pollable Data: '+ JSON.stringify(twitchData));
+    	socket.emit('update twitch readonly data', twitchData);
+    }
+
+    function getTwitchFollowerData(twitchData) {
+    	var requestUrl = config.twitchApiRoot + '/channels/'+twitchData.twitchUsername+'/follows/';
+    	var xmlhttp = new XMLHttpRequest();
+    	xmlhttp.open('GET',requestUrl,false);
+  		xmlhttp.setRequestHeader('Accept','application/vnd.twitchtv.v2+json');
+  		xmlhttp.setRequestHeader('Client-Id', config.twitchClientId);
+  		xmlhttp.send();
+  		console.log('Requesting follower data for '+twitchData.twitchUsername);
+  		var twitchResponse = JSON.parse(xmlhttp.responseText);
+  		twitchData.twitchFollowers = twitchResponse._total;
+  		twitchData.twitchLastFollower = twitchResponse.follows[0].user.name;
+  		return twitchData;
+    }
+
+    function getTwitchViewerData(twitchData) {
+      var requestUrl = config.twitchApiRoot + '/streams/'+twitchData.twitchUsername;
+      var xmlhttp = new XMLHttpRequest();
+      xmlhttp.open('GET',requestUrl,false);
+      xmlhttp.setRequestHeader('Accept','application/vnd.twitchtv.v2+json');
+      xmlhttp.setRequestHeader('Client-Id', config.twitchClientId);
+      xmlhttp.send();
+      console.log('Requesting viewer data for '+twitchData.twitchUsername);
+      var twitchResponse = JSON.parse(xmlhttp.responseText);
+      var stream = twitchResponse.stream;
+      twitchData.twitchViewers = stream ? stream.viewers : null;
+      if (twitchData.twitchViewers)
+      {
+        if(!twitchData.twitchPeakViewers || twitchData.twitchViewers > twitchData.twitchPeakViewers)
+        {
+          twitchData.twitchPeakViewers = twitchData.twitchViewers;
+        }
+      }
+      return twitchData;
+    }
+
+    function updateTwitchData(twitchData) {
+    	var channelName = twitchData.twitchUsername;
+    	var game = setTwitchGame(twitchData.twitchUsername, twitchData.twitchGame).game;
+      var status = setTwitchStatus(twitchData.twitchUsername, twitchData.twitchStatus).status;
+      twitchData.twitchGame = game;
+      twitchData.twitchStatus = status;
+      return twitchData;
```

Comment:
Git isn't acknowledging the whitespace change from tabs to spaces, but hopefully they'll get pulled in during a different CR change.